electrum/util.py

```diff
@@ -1952,12 +1952,16 @@ async def open_connection(self, host=None, port=None, **kwargs):
         return reader, writer
 
     @classmethod
-    def from_proxy_dict(cls, proxy: dict = None) -> Optional['MySocksProxy']:
-        if not proxy:
+    def from_network_settings(cls, network: Optional['Network']) -> Optional['MySocksProxy']:
+        if not network or not network.proxy:
             return None
+        proxy = network.proxy
         username, pw = proxy.get('user'), proxy.get('password')
         if not username or not pw:
-            auth = None
+            if network.is_proxy_tor:
+                auth = aiorpcx.socks.SOCKSRandomAuth
+            else:
+                auth = None
         else:
             auth = aiorpcx.socks.SOCKSUserAuth(username, pw)
```

Comment:
Notice how auth have different types here. This should raise red flags in the reader's mind.

Indeed the random auth is not working. aiorpcx has a bad implementation here, it should really really raise an exception instead of silently ignoring the random auth.

https://github.com/kyuupichan/aiorpcX/blob/05ab957077f961db7ab4acd3f06c690d63eda6ab/aiorpcx/socks.py#L184
https://github.com/kyuupichan/aiorpcX/blob/05ab957077f961db7ab4acd3f06c690d63eda6ab/aiorpcx/socks.py#L201

You need `auth = aiorpcx.socks.SOCKSRandomAuth()`
not `auth = aiorpcx.socks.SOCKSRandomAuth`

The second one is a class, the first one is an instance of the class.

You can test with
```patch
diff --git a/aiorpcx/socks.py b/aiorpcx/socks.py
index 4a6cefa..2dc3da7 100755
--- a/aiorpcx/socks.py
+++ b/aiorpcx/socks.py
@@ -182,6 +182,7 @@ class SOCKS5(SOCKSBase):
         super().__init__()
         self._dst_bytes = SOCKS5._destination_bytes(remote_address.host, remote_address.port)
         self._auth_bytes, self._auth_methods = SOCKS5._authentication(auth)
+        print(f"{self._auth_bytes=}, {self._auth_methods=}")
 
     @staticmethod
     def _destination_bytes(host, port):
```

Reply:
Yes I know the difference. Nice that it is silently ignored :)